base/system/winlogon/sas.c

```diff
@@ -1070,12 +1076,15 @@ DoGenericAction(
                     LogOffFlags |= EWX_FORCE;
                 if (!Session->Gina.Functions.WlxIsLogoffOk(Session->Gina.Context))
                     break;
-                if (!NT_SUCCESS(HandleLogoff(Session, LogOffFlags)))
+                if (NT_SUCCESS(HandleLogoff(Session, LogOffFlags)))
                 {
-                    RemoveStatusMessage(Session);
-                    break;
+                    Session->Gina.Functions.WlxLogoff(Session->Gina.Context);
+                    if (!WLX_SHUTTINGDOWN(wlxAction))
+                    {
+                        RemoveStatusMessage(Session);
+                        Session->Gina.Functions.WlxDisplaySASNotice(Session->Gina.Context);
+                    }
```

Comment:
Can you explain (here in the PR only, first) why this reorganization is necessary?

Reply:
Without reorganizing it, it would be impossible to shut down from an account that fails to log off (for example, the system account.)

Reply:
ok I'll try to understand more this change.